module/rdpClientCon.c

```diff
@@ -839,6 +839,14 @@ rdpClientConProcessMsgClientInfo(rdpPtr dev, rdpClientCon *clientCon)
     memcpy(&(clientCon->client_info), s->p - 4, bytes);
     clientCon->client_info.size = bytes;
 
+    if (clientCon->client_info.version != CLIENT_INFO_CURRENT_VERSION)
+    {
+        LLOGLN(0, ("expected xrdp client_info version %d, got %d",
+                   CLIENT_INFO_CURRENT_VERSION,
+                   clientCon->client_info.version));
+        FatalError("Incompatible xrdp version detected  - please recompile");
```

Comment:
I'm not familiar with `FatalError`, what exactly does  do?

Can the xorgxrdp version number for the client info be sent back to xrdp as a separate message or a disconnect error code? Having the disconnect error code means that the error can be communicated back to the user attempting to login, instead of being in the xorgxrdp log file which is inaccessible to the user.

Reply:
FatalError is an internal Xorg call. It produces the 'EE' lines in the log file :-

```
[   289.083] (EE) Incompatible xrdp version detected  - please recompile(EE)
[   289.083] (EE)
```

I've added a comment below about your disconnect error code.